functions/api-router/index.js

```diff
@@ -0,0 +1,57 @@
+const functions = require('firebase-functions');
+const admin = require('firebase-admin');
+admin.initializeApp();
+
+const db = admin.database();
+
+exports.api_router = functions.https.onRequest(async (req, res) => {
+  const itemId = req.query.itemId || req.body.itemId;
+  const itemData = req.body.data;
+
+  switch (req.method) {
+    case 'POST':
```

Comment:
not fully following here -- is this just one API endpoint and we're doing different actions on it?

i.e. `POST https://https://scl-scavengerhunt.firebaseio.com`, do we not have any routes like `/users/:id`?

Also totally fine if this is just scaffolding / examples for now!

Reply:
this is a scaffolding action/ I believe this is one API endpoint, im unsure how the routing works ATM.

I believe this is only the `functions` subdomain. I believe the origin will be scl-scavengerhunt.firebase.app etc right now so we won't have to do complex URL routing. 

we can also add aliases. This is a good question and todo if we want to flesh out an endpoint design doc